easybuild/easyblocks/c/crisprdav.py

```diff
@@ -0,0 +1,100 @@
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Metagenome-Atlas, implemented as an easyblock.
+
+@author: Denis Kristak (INUITS)
+"""
+import os
+from easybuild.tools.filetools import write_file
+from easybuild.easyblocks.generic.binary import Binary
+
+
+class EB_crisprdav(Binary):
+    """
+    Support for building/installing crispr-dav.
+    """
+
+    def post_install_step(self):
+        """Create config.txt files"""
+        crisprdav_installdir_ex = os.path.join(self.installdir, 'Examples/example1/')
+        abra2_dir = os.environ['EBROOTABRA2']
+        prinseq_dir = os.environ['EBROOTPRINSEQ']
+        flash_dir = os.environ['EBROOTFLASH']
+
+        config_file = os.path.join(self.installdir, 'conf.txt')
+        example_fastq_list_file = os.path.join(self.installdir, 'Examples/example1/fastq.list')
+
+        # writing to Example directory to run sanity checks
+        example_config_file = os.path.join(self.installdir, 'Examples/example1/conf.txt')
+
+        CONFIG_FILE_TEXT_FORMATTED = CONFIG_FILE_TEXT.format(crisprdav_installdir_ex=crisprdav_installdir_ex,
+                                                             abra2_dir=abra2_dir, prinseq_dir=prinseq_dir,
+                                                             flash_dir=flash_dir)
+        FASTQ_LIST_FILE_TEXT_FORMATTED = FASTQ_LIST_FILE_TEXT.format(crisprdav_installdir_ex=crisprdav_installdir_ex)
+        # according to docs, we have to setup conf.txt so that it contains correct paths to dependencies
+        # https://github.com/pinetree1/crispr-dav/blob/master/Install-and-Run.md
+        # User then has to change conf.txt to include paths to genomes
+        write_file(config_file, CONFIG_FILE_TEXT_FORMATTED)
+        write_file(example_config_file, CONFIG_FILE_TEXT_FORMATTED)
+        # also used for sanity checking with Examples/ folder
+        write_file(example_fastq_list_file, FASTQ_LIST_FILE_TEXT_FORMATTED)
+
+
+# obtained from
+# https://github.com/pinetree1/crispr-dav/blob/master/conf.txt
+# see the original file for detailed commentary
+CONFIG_FILE_TEXT = """
+# change this part to contain paths to your genomes
+[genomex]
+ref_fasta = {crisprdav_installdir_ex}/genome/genomex.fa
+bwa_idx = {crisprdav_installdir_ex}/genome/genomex.fa
+refGene = {crisprdav_installdir_ex}/genome/refgenex.txt
+
+[app]
+abra = {abra2_dir}/abra2-2.23.jar
```

Comment:
Wouldn't it be better to copy the `conf.txt`, and do a couple of search-and-replace operations on that?

Shipping a copy of the `conf.txt` in this easyblock makes it a maintenance problem going forward...